rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -242,11 +252,57 @@ class PlayerImpl
     std::shared_ptr<rclcpp::GenericPublisher> publisher_;
     std::function<void(const rclcpp::SerializedMessage &)> publish_func_;
   };
+
+  class PlayerClient final
```

Comment:
Comments from MichaelOrlov

Please rename PlayerClient to the PlayerServiceClient and consider moving it to the separate cpp and hpp files.
The PlayerClient is pretty large and independent from its parent class. We already have a lot of code inside the Player class and with this inner class inside it will become more cluttered and more difficult to navigate and support.

Reply:
Yes. I will move PlayerServiceClient out of PlayerImpl class.